agents/labjackT7

```diff
@@ -0,0 +1,215 @@
+    data[1] = intNum & 0xFFFF
+    return data
+
+
+'''
+Converting data arrays to numbers
+'''
+
+def data_to_uint16(data):
+    return data[0]
+
+
+def data_to_uint32(data):
+    return concatData(data)
+
+
+def data_to_int32(data):
+    return struct.unpack("=i", struct.pack("=I", concatData(data)))[0]
+
+
+def data_to_float32(data):
+    return struct.unpack("=f", struct.pack("=I", concatData(data)))[0]
+
+
+
+'''
+LabJack agent class
+'''
+
+class LabJackT7_agent:
+    def __init__(self,agent,ip_address,num_channels):
+        self.active = True
+        self.agent = agent
+        self.ip_address = ip_address
+        self.module = None
+        self.log = agent.log
+        self.lock = TimeoutLock()
+        self.log = agent.log
+        self.sensors = ['Channel {}'.format(i + 1) for i in range(num_channels)]
+        
+        self.initialized = False
+        self.take_data = False
+        
+        # Register feed
+        agg_params = {
+            'frame_length': 60,
+        }
+        self.agent.register_feed('Sensors',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=1)
+
+
+    def init_labjack_task(self, session, params = None):
+        """
+        task to initialize labjack module
+        """
+        
+        if self.initialized:
+            return True, "Already initialized module"
+        
+        with self.lock.acquire_timeout(0, job = 'init') as acquired:
+            if not acquired:
+                self.log.warn("Could not start init because" "{} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+    
+            session.set_status('starting')
+        
+        self.module = ModbusTcpClient(str(self.ip_address))
+        print("Initialized labjack module")
+        session.add_message("Labjack initialized")
+    
+        self.initialized = True
+        return True, 'LabJack module initialized.'
+    
+    
+    
+    def start_acq(self, session, params=None):
+        """
+        Task to start data acquisition.
+        Args:
+            sampling_frequency (float):
+                Sampling frequency for data collection. Defaults to 2.5 Hz
+        """
+        if params is None:
+            params = {}
+
+        f_sample = params.get('sampling_frequency', 2.5)
+        sleep_time = 1/f_sample - 0.01
+
+        with self.lock.acquire_timeout(0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn("Could not start acq because {} is already running"
+                              .format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('running')
+
+            self.take_data = True
+
+            while self.take_data:
+                data = {
+                    'timestamp': time.time(),
+                    'block_name': 'temps',
```

Comment:
The block name should be changed to something that identifies what the data being collected with the LabJack is, or maybe even just that it's the labjack taking the data. 'temps' appears to be a hold over from the 240 Agent.